### PR TITLE
feat: add new apispec for the new amalthea sessions

### PR DESCRIPTION
**Pull request by olevski**
This will be used only in Renku v2 with the new Amalthea operator.

### CODE REVIEW

File: components/renku_data_services/notebooks/api.spec.yaml [hunk 1]
```diff
@@ -723,6 +876,158 @@ components:
       - renku.io/projectName
       - renku.io/repository
       type: object
+    SessionPostRequest:
+      properties:
+        project_id:
+          $ref: "#/components/schemas/Ulid"
+        launcher_id:
+          $ref: "#/components/schemas/Ulid"
+        storage:
```
Comment (andre-code) [inline components/renku_data_services/notebooks/api.spec.yaml:885]: Consider renaming this to disk_storage for better clarity.

File: components/renku_data_services/notebooks/api.spec.yaml [hunk 2]
```diff
@@ -723,6 +876,158 @@ components:
       - renku.io/projectName
       - renku.io/repository
       type: object
+    SessionPostRequest:
```
Comment (andre-code) [inline components/renku_data_services/notebooks/api.spec.yaml:879]: Allow sending cloud storage details `cloudstorage ` , as it's useful when storage credentials are missing.

File: components/renku_data_services/notebooks/api.spec.yaml [hunk 3]
```diff
@@ -723,6 +876,158 @@ components:
...
+          description: The size of disk storage for the session, in gigabytes
+        resource_class_id:
+          default: 
+          nullable: true
+          type: integer
+      required:
+        - project_id
+        - launcher_id
+      type: object
+    SessionResponse:
```
Comment (andre-code) [inline components/renku_data_services/notebooks/api.spec.yaml:897]: Which part here represents the sessionId?

**Review by andre-code — CHANGES_REQUESTED** [review]
The API specification looks good overall. Aside from my comment below, it handles all the cases currently used for Renku 2.0.

File: components/renku_data_services/notebooks/api.spec.yaml (hunk 3 above)
Comment (olevski) [reply components/renku_data_services/notebooks/api.spec.yaml:897]: It's the `name` field.

**Comment by olevski** [comment]
This change is part of the following stack:

- #360 ◀

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>

File: components/renku_data_services/notebooks/api.spec.yaml (hunk 2 above)
Comment (olevski) [reply components/renku_data_services/notebooks/api.spec.yaml:879]: Done.

File: components/renku_data_services/notebooks/api.spec.yaml (hunk 1 above)
Comment (olevski) [reply components/renku_data_services/notebooks/api.spec.yaml]: Done

**Review by Panaetius — APPROVED** [review]

**Pull request merged**